src/pages/Home/Home.test.js

```diff
@@ -0,0 +1,338 @@
+}));
+
+jest.mock('firebase/dailyResults', () => ({
+  addDailyResults: () => '1234567',
+  getDailyResults: () => ({
+    docs: [],
+    error: false,
+  }),
+  updateDailyResults: jest.fn(),
+}));
+
+jest.mock('firebase/usersStatistics', () => ({
+  getUsersStatistics: () => ({
+    currentStatistics: emptyStatistics,
+  }),
+  updateUsersStatistics: () => {},
+}));
+
+const getKeyboardLetter = letter => screen.getByRole('button', { name: letter });
+
+const getLetters = buttons =>
+  buttons.filter(item => {
+    const ariaRoleDescription = item.getAttribute('aria-roledescription');
+    return ariaRoleDescription === 'letter';
+  });
+
+beforeEach(async () => {
+  render(<Home />, {
+    preloadedState: {
+      user: {
+        user: {
+          email: 'janedoe@rootstrap.com',
+          name: 'Jane Doe',
+          photo: 'https://source.unsplash.com/random/?person',
+          uid: 'ABCDEFGHIJK',
+        },
+        authenticated: true,
+      },
+    },
+  });
+
+  await waitFor(() => {
+    const emptyWord = screen.getAllByRole('group', { name: 'empty' });
+    expect(emptyWord).toHaveLength(1);
+
+    const emptyLetters = screen.getAllByRole('button', { name: 'empty' });
+    expect(emptyLetters).toHaveLength(5);
+  });
+});
+
+test('enter all incorrect letters (CLASS)', async () => {
+  // get all the keyboard buttons we will use
+  const CButton = getKeyboardLetter('C');
+  const LButton = getKeyboardLetter('L');
+  const AButton = getKeyboardLetter('A');
+  const SButton = getKeyboardLetter('S');
+
+  // type CLASS with the keyboard buttons
+  fireEvent.click(CButton);
+  fireEvent.click(LButton);
+  fireEvent.click(AButton);
+  fireEvent.click(SButton);
+  fireEvent.click(SButton);
+
+  // check there are no more empty spaces and now it has CLASS content
+  const emptyWord = screen.queryByRole('group', { name: 'empty' });
+  expect(emptyWord).not.toBeInTheDocument();
+
+  const classWord = screen.queryByRole('group', { name: 'CLASS' });
+  expect(classWord).toBeInTheDocument();
+
+  const emptyLetters = screen.queryByRole('button', { name: 'empty' });
+  expect(emptyLetters).not.toBeInTheDocument();
+
+  const CLetters = screen.getAllByRole('button', { name: 'C' });
+  expect(getLetters(CLetters)).toHaveLength(1);
+
+  const LLetters = screen.getAllByRole('button', { name: 'L' });
+  expect(getLetters(LLetters)).toHaveLength(1);
+
+  const ALetters = screen.getAllByRole('button', { name: 'A' });
+  expect(getLetters(ALetters)).toHaveLength(1);
+
+  const SLetters = screen.getAllByRole('button', { name: 'S' });
+  expect(getLetters(SLetters)).toHaveLength(2);
+
+  // submit the word
+  const enterButton = screen.getByRole('button', { name: /enter/i });
+  fireEvent.click(enterButton);
+
+  await waitFor(() => {
+    // the CLASS letter's row and CLAS keys has all absent aria label (9 in total)
+    const absentLetters = screen.getAllByRole('button', { name: /absent/i });
+    expect(absentLetters).toHaveLength(9);
```

Comment:
to make it clear why should be 9 you can divide this into two separate checks using `within`:  

'CLASS' can be on a constant  and then: 

```
  const word = screen.getByRole('group', { name: CLASS });
  expect(within(word).getAllByRole('button', { name: /absent/i })).toHaveLength(CLASS.length);

  const keyboard = screen.getByRole('group', { name: 'keyboard'});
  expect(within(keyboard).getAllByRole('button', { name: /absent/i })).toHaveLength(4);
```

Reply:
Awesome, thanks!!! I didn't know about `within` and its great!